bciers/apps/registration/tests/components/form/widgets/PostalCodeWidget.test.tsx

```diff
@@ -0,0 +1,159 @@
+  it("should render a postal code field", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+    expect(screen.getByLabelText(postalCodeLabelRequired)).toBeVisible();
+  });
+
+  it("should be empty by default", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+
+    const input = screen.getByLabelText(postalCodeLabelRequired);
+
+    expect(input).toHaveValue("");
+  });
+
+  it("should render the postal code value when formData is provided", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        formData={{ postalCodeTestField: "A1B2C3" }}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+
+    const input = screen.getByLabelText(postalCodeLabelRequired);
+    expect(input).toHaveValue(postalCode);
+  });
+
+  it("should allow entering a postal code", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+    const input = screen.getByLabelText(postalCodeLabelRequired);
+    await userEvent.type(input, postalCode);
+    expect(input).toHaveValue(postalCode);
+  });
+
+  it("should convert lowercase postal code to uppercase", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+    const input = screen.getByLabelText(postalCodeLabelRequired);
+    await userEvent.type(input, "a1b 2c3");
+    expect(input).toHaveValue(postalCode);
+  });
+
+  it("should allow entering a postal code with no spaces", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+    const input = screen.getByLabelText(postalCodeLabelRequired);
+    await userEvent.type(input, "A1B2C3");
+    expect(input).toHaveValue(postalCode);
+  });
+
+  it("should trigger validation error for required field", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+    const submitButton = screen.getByRole("button", { name: "Submit" });
+
+    await userEvent.click(submitButton);
+
+    expect(screen.getByText("Required field")).toBeVisible();
+  });
+
+  it("should trigger validation error for invalid postal code", async () => {
+    render(
+      <FormBase
+        schema={postalCodeFieldSchema}
+        uiSchema={postalCodeFieldUiSchema}
+      />,
+    );
+    const input = screen.getByLabelText(postalCodeLabelRequired);
+    await userEvent.type(input, "A1B2CC");
+
+    const submitButton = screen.getByRole("button", { name: "Submit" });
+
+    await userEvent.click(submitButton);
+
+    expect(screen.getByText("Format should be A1A 1A1")).toBeVisible();
```

Comment:
Horrible thought: if we're accepting American phone numbers, should we also be accepting American zip codes...? (This thought makes me sad.)

Reply:
:scream:  I'll ask the designers